tracee/tracee.bpf.c

```diff
@@ -497,13 +500,31 @@ static __always_inline int get_config(u32 key)
     return *config;
 }
 
+// returns 1 if you should trace based on uid, 0 if not
 static __always_inline int uid_filter_matches()
 {
     uid_t uid = (u32)bpf_get_current_uid_gid();
-    u32* isMatch = bpf_map_lookup_elem(&uid_filter, &uid);
-    if (isMatch == NULL) {
-        return 0;
+    s64* equality = bpf_map_lookup_elem(&uid_equal_filter, &uid);
+    if (equality != NULL) {
+        return *equality;
+    }
+
+    u32 greater = GREATER;
+    u32 less = LESS;
+
+    s64* greaterThan = bpf_map_lookup_elem(&uid_compare_filter, &greater);
+    s64* lessThan = bpf_map_lookup_elem(&uid_compare_filter, &less);
+
+    if (lessThan != NULL) {
```

Comment:
Just for aesthetic, I think we can have one "if" here (as there is no verifier issue with this):
if ((lessThan != NULL) && (uid >= *lessThan))